src/server/pkg/storage/fileset/reader.go

```diff
@@ -0,0 +1,30 @@
+package fileset
+
+import (
+	"archive/tar"
+	"io"
+)
+
+// Reader is a wrapper for a tar reader that interprets
+// Pachyderm specific flags (particularly MergeType).
+type Reader struct {
+	r tar.Reader
+}
+
+func NewReader(rs []io.Reader) *Reader {
```

Comment:
I think this should probably take a single `io.Reader`, and then that reader can specify subreaders that are being merged. Doing it this way introduces some minor, but I think long-term impactful, fragmentation. Because we can have a merge of multiple Filesets specified both within the code, by passing multiple readers to this function, and within the data when a Fileset references another one. In the latter case we're likely going to have more information, such as bloom filters for the merged sets, and bounding keys that we don't have in the former case. Unifying this early on should help keep the design simple.

Reply:
Yeah, that makes sense. This stub was from early on, and I have not been working on it yet. I am still trying to figure out what this actually means in terms of implementation though.